framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/OpenShiftDeployment.java

```diff
@@ -106,15 +108,27 @@ private Instance createInstance(Pod pod) {
     }
 
     protected URL getHttpRouteUrl(String serviceName) {
-        return getRouteUrl("http", serviceName);
+        try {
+            return getRouteUri("http", serviceName).toURL();
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     protected URL getHttpsRouteUrl(String serviceName) {
-        return getRouteUrl("https", serviceName);
+        try {
+            return getRouteUri("https", serviceName).toURL();
```

Comment:
serviceName is here name of the service or name of the route? Because route for https has prefix "secure-" (e.g. [here](https://github.com/jboss-openshift/application-templates/blob/bpmsuite-wip/bpmsuite/bpmsuite70-full-mysql.json#L476) ). On the other hand, I am not sure, if this is same for all templates.

Reply:
It is a name of the service. Cloud framework expects that service name is equal to route name.